libsql/src/local/database.rs

```diff
@@ -402,7 +402,7 @@ impl Database {
 
         let max_frame_no = conn.wal_frame_count();
 
-        let generation = 1; // TODO: Probe from WAL.
+        let generation = sync_ctx.generation(); // TODO: Probe from WAL.
         let start_frame_no = sync_ctx.durable_frame_num() + 1;
```

Comment:
Now that we cache durable frame number, we need to be prepared for the scenario where remote database got _deleted_. IOW. the durable frame number we have here might be higher than what is actually on server side.